libr/core/cio.c

```diff
@@ -425,44 +390,18 @@ R_API int r_core_shift_block(RCore *core, ut64 addr, ut64 b_size, st64 dist) {
 	return res;
 }
 
-static RCoreFile * r_core_file_set_first_valid(RCore *core) {
-	RListIter *iter;
-	RCoreFile *file = NULL;
-
-	r_list_foreach (core->files, iter, file) {
-		if (file && file->desc) {
-			core->io->raised = file->desc->fd;
-			core->switch_file_view = 1;
-			break;
-		}
-	}
-	return file;
-}
-
 R_API int r_core_block_read(RCore *core) {
-	if (!core->file && !r_core_file_set_first_valid (core)) {
-		memset (core->block, core->io->Oxff, core->blocksize);
-		return -1;
+	if (core && core->block) {
+		return r_io_read_at (core->io, core->offset, core->block, core->blocksize);
 	}
-	if (core->file && core->switch_file_view) {
-		r_io_use_desc (core->io, core->file->desc);
-		r_core_bin_set_by_fd (core, core->file->desc->fd); //needed?
-		core->switch_file_view = 0;
-	} 
-	return r_io_read_at (core->io, core->offset, core->block, core->blocksize);
+	return -1;
 }
 
 R_API int r_core_read_at(RCore *core, ut64 addr, ut8 *buf, int size) {
```

Comment:
Should return bool imho. Core reads can be across multiple sections and such, so it will not return any amount of read bytes. Better stick those high level apis to bool